cpu/stm32l1/periph/i2c.c

```diff
@@ -85,27 +81,31 @@ int i2c_init_master(i2c_t dev, i2c_speed_t speed)
             return -2;
     }
 
-    /* read static device configuration */
     switch (dev) {
-#if I2C_0_EN
+
         case I2C_0:
-            i2c = I2C_0_DEV;
-            port = I2C_0_PORT;
-            pin_scl = I2C_0_SCL_PIN;
-            pin_sda = I2C_0_SDA_PIN;
-            I2C_0_CLKEN();
             I2C_0_PORT_CLKEN();
-            NVIC_SetPriority(I2C_0_ERR_IRQ, I2C_IRQ_PRIO);
-            NVIC_EnableIRQ(I2C_0_ERR_IRQ);
             break;
-#endif
+
+        case I2C_1:
+            I2C_1_PORT_CLKEN();
+            break;
 
         default:
             return -1;
     }
 
+    /* enable I2C clock */
+    i2c_poweron(dev);
+
+    /* set IRQn priority */
+    NVIC_SetPriority(i2c_config[dev].er_irqn, I2C_IRQ_PRIO);
+
+    /* enable IRQn */
+    NVIC_EnableIRQ(i2c_config[dev].er_irqn);
+
     /* configure pins */
-    _pin_config(port, pin_scl, pin_sda);
+    _pin_config(dev);
```

Comment:
@ReneHerthel please look at the GPIO API. I'd like to see it being used instead of reprogramming IO- and AF-initialization. Can you check that please? In that case you can exchange the whole ` _pin_config` function by some GPIO API calls